core/lib/server/stages.ts

```diff
@@ -48,33 +49,27 @@ export const getPubIdsInStage = (stageId: StagesId) =>
 			.where("stageId", "=", stageId)
 	);
 
-export const getCommunityStages = (communityId: CommunitiesId) =>
+type CommunityStageProps = XOR<{ communityId: CommunitiesId }, { stageId: StagesId }>;
+export const getCommunityStages = ({ communityId, stageId }: CommunityStageProps) =>
```

Comment:
i like this change, but could we maybe change the name a bit? Maybe like `getStages`?

Also, maybe require passing the `communityId` anyway, that way it's less likely for us to accidentally allow fetching other communities' stage info by doing something like `/c/${your_community_slug}/stages/${someone_elses_stage}`

I did the same for the `getPubsWith...` function for the same reason

Reply:
two great ideas, ty!